### PR TITLE
Updated some examples to use the new plotting backend

### PR DESCRIPTION
**Pull request by AayushSabharwal**
* Daisyworld
* Game of Life
* HK opinion dynamics
* Predator-Prey

### CODE REVIEW

**Comment by Libbum** [comment]
Excellent. So we can merge this into cairomakie a little later. For now, feel free to continue updating examples and we'll merge once the list is done. I've got other things on my plate today so most likely wont be adding anything for a while. 

If you keep editing your list above, and just put whatever example you're working on with *(in progress)* next to it, I'll know not to touch it.

File: examples/daisyworld.jl [hunk 1]
```diff
@@ -49,10 +49,10 @@
 # visualize surface temperature.
 # It is also available from the `Models` module as [`Models.daisyworld`](@ref).
 
-using Agents, AgentsPlots, Plots
+using Agents
 using Statistics: mean
 using Random # hide
-gr() # hide
+# gr() # hide
```
Comment (Libbum) [inline examples/daisyworld.jl:55]: We don't need this at all, feel free to delete the line.

File: examples/daisyworld.jl [hunk 2]
```diff
@@ -255,9 +255,11 @@ nothing # hide
 
 cd(@__DIR__) #src
 Random.seed!(165) # hide
+using InteractiveChaos, CairoMakie
+using Makie # hide
```
Comment (Libbum) [inline examples/daisyworld.jl:259]: Instead of Makie, just use AbstractPlotting here. Also, no need to hide it.

File: examples/daisyworld.jl [hunk 3]
```diff
@@ -318,11 +315,12 @@ Random.seed!(165) # hide
...
-p = plot(agent_df[!, :step], agent_df[!, :count_black_daisies], label = "black")
-plot!(p, agent_df[!, :step], agent_df[!, :count_white_daisies], label = "white")
-plot!(p; xlabel = "tick", ylabel = "daisy count")
-
+figure = Figure()
+ax = figure[1, 1] = Axis(figure, xlabel="tick", ylabel="daisy count")
+blackl = lines!(ax, agent_df[!, :step], agent_df[!, :count_black_daisies], color=:red)
+whitel = lines!(ax, agent_df[!, :step], agent_df[!, :count_white_daisies], color=:blue)
+figure[1, 2] = Legend(figure, [blackl, whitel], ["black", "white"], textsize=12)
+Makie.save("figure1.png", figure) # hide
```
Comment (Libbum) [inline examples/daisyworld.jl:323]: Remove the save line, Documenter.jl will display the plot in-line so long as the last line in this block is `figure`.

File: examples/game_of_life_2D_CA.jl [hunk 4]
```diff
@@ -90,17 +89,15 @@ end
...
-    p1 = plotabm(model; ac = ac, as = 3, am = :square, showaxis = false)
-end
+am(x) = x.status == true ? '■' : '□'
+abm_video(
+    "game of life.mp4", model, dummystep, ca_step!;
+    title="Game of Life", ac=:black, as=12,
+    am = am, framerate=5
+)
 nothing # hide
-
```
Comment (Libbum) [inline examples/game_of_life_2D_CA.jl:102]: If you take a look at some of the files I've updated, after the `abm_video` call we need to put
```julia
nothing # hide
# ```@raw html
# <video width="auto" controls autoplay loop>
# <source src="../file_name.mp4" type="video/mp4">
# </video>
# ```
```

Additionally, under the first heading in most examples, there's a link to the gif file. That can be replaced by the raw html block as well.

File: examples/hk.jl [hunk 5]
```diff
@@ -142,22 +142,29 @@ agent_data, _ = run!(
...
-
-plotsim(data, ϵ) = plot(
-    data.step,
-    data.new_opinion,
-    leg = false,
-    group = data.id,
-    title = "epsilon = $(ϵ)",
-)
 
-Random.seed!(42) # hide
```
Comment (Libbum) [inline examples/hk.jl:157]: The seeds are useful so we don't have the documentation 'look' changing everytime we alter something in the repository. We need to keep it.

File: examples/predator_prey.jl [hunk 6]
```diff
@@ -29,9 +29,9 @@
 
 # It is also available from the `Models` module as [`Models.predator_prey`](@ref).
 
-using Agents, AgentsPlots, StatsPlots
+using Agents
 using Random # hide
-pyplot() # hide
+# pyplot() # hide
```
Comment (Libbum) [inline examples/predator_prey.jl:34]: Can be removed.

File: examples/predator_prey.jl [hunk 7]
```diff
@@ -244,17 +245,13 @@ results, _ = run!(model, agent_step!, n_steps; adata)
...
-@df results plot(
-    :step,
-    :count_sheep,
-    grid = false,
-    xlabel = "Step",
-    ylabel = "Population",
-    label = "Sheep",
-)
-@df results plot!(:step, :count_wolves, label = "Wolves")
-@df results plot!(:step, :count_grass, label = "Grass")
```
Comment (Libbum) [inline examples/predator_prey.jl:257]: An additional note for us devs: we can also remove the StatsPlots dependency in the docs with this change.

Comment (Datseris) [reply examples/predator_prey.jl:257]: ok

**Comment by Libbum** [comment]
@AayushSabharwal, when you have implemented changes from the review, click on the "Resolve conversation" buttons of the things you changed, so we can keep track of what hasn't been fixed yet.

**Comment by Libbum** [comment]
SIR cannot be updated (easily) since there are no graph recipies in Makie yet. Flocking and Bacteria Growth we are working on getting patches in `scatter` with the devs, but not sure how long that will be.

It's the integration examples that are left now.

**Comment by Libbum** [comment]
Thanks for your work here @AayushSabharwal! I'm going to merge this in soon and then do some final alterations in the `cairomakie` branch. 

Unsure when we'll merge the lot since we have some open questions concerning a few of the examples.

**Pull request merged**